sdk/core/Azure.Core/src/AsyncCollection.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for
+// license information.
+
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Threading;
+
+namespace Azure
+{
+    /// <summary>
+    /// A collection of values that may take multiple service requests to
+    /// iterate over.
+    /// </summary>
+    /// <typeparam name="T">The type of the values.</typeparam>
+    public abstract class AsyncCollection<T> : IAsyncEnumerable<Response<T>>
```

Comment:
Open question - are we going to have a `SyncCollection<T>` or similar as an indicator to customers that the sync version of `GetBlobs()` may make 100 requests?